### PR TITLE
fix issue where java parses class incorrectly from lambdas after new

### PR DESCRIPTION
**Pull request by dmoody256**
Found a small issue related to parsing java files. I assume this was never run into in earlier versions of java because lambda was added in java 1.8.

With out the small change to see lambda tokens, the java parser would skip over that notation and think a anonymous class was being created if a new was used just prior. This change make the java parser aware of the lambda notation.

See the supplied test case for more info.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: SCons/Tool/JavaCommon.py [hunk 1]
```diff
@@ -88,7 +88,7 @@
     #     the multi-line comment begin and end tokens /* and */;
     #     array declarations "[]".
     _reToken = re.compile(r'(\n|\\\\|//|\\[\'"]|[\'"{\};.()]|' +
-                          r'\d*\.\d*|[A-Za-z_][\w$.]*|<[A-Za-z_]\w+>|' +
+                          r'\d*\.\d*|[A-Za-z_][\w$.]*|->|<[A-Za-z_]\w+>|' +
```
Comment (bdbaddog) [inline SCons/Tool/JavaCommon.py]: Please update the comment block above with the change

File: SCons/Tool/JavaCommonTests.py [hunk 2]
```diff
@@ -484,6 +484,41 @@ def test_nested_anonymous_inner_classes(self):
...
+{
+
+    public void testFunc (int arg1, String arg2, Runnable lambda){
+    }
+    public LamdaExample()
+    {
+        testFunc(
+            5,
+            new String("test"),
+            () -> {}
```
Comment (bdbaddog) [inline SCons/Tool/JavaCommonTests.py:505]: Stick a comment above the lambda to indicate that's what it is? (Not everyone's familiar with Java lambda syntax)

**Comment by bdbaddog** [comment]
A couple minor nits, but otherwise looks good.
Once you fix those I'll be able to merge.

File: SCons/Tool/JavaCommonTests.py (hunk 2 above)
Comment (dmoody256) [reply SCons/Tool/JavaCommonTests.py:505]: updated in 1b2e8e9

File: SCons/Tool/JavaCommon.py (hunk 1 above)
Comment (dmoody256) [reply SCons/Tool/JavaCommon.py]: updated in 1b2e8e9

**Comment by bdbaddog** [comment]
Great!
Thanks.
Merging.

**Pull request merged**